web/src/components/topology/Korrel8rTopology.tsx

```diff
@@ -284,7 +284,7 @@ export const Korrel8rTopology: React.FC<{
     setTimeout(() => {
       // Center the graph on the next render tick once the graph elements have been sized and placed
       controller.getGraph().fit(30);
-    }, 1);
+    }, 100); // FIXME this timeout is racy, need to reliably centre graph after layout is done.
```

Comment:
I had tried to get the center graph to work as we load it in, but wasn't able to figure it out.

As a touch of a hacky solution (my favorite kind) we could leave the timeout here, and then put a slightly longer timeout on the "loading" state and make the "loading" state sit higher on the page than loaded state. That way the loading state shows while the flash occurs underneath it.

Outside of that, we would need to contact someone with experience using the Topology view, as it is very complex and doesn't have the best documentation.

I'd say to leave that issue to a future PR and leave this one as it is for now if that is alright.

Reply:
OK, I'll try a bit more digging - there has to be a way to set a sensible initial layout without arbitrary timeouts. The 100ms timeout seems to work most of the time, but there is a visible flicker.

Reply:
I found a solution in slack: https://redhat-internal.slack.com/archives/C04JMHKSD9C/p1727125692797519

Remove the timeout and add `controller.setFitToScreenOnLayout(true);`

Reply:
Tried it, didn't work. I'd like to merge the other fixes in this PR now (with the longer 100ms timeout, which does the job with a bit of flicker)  and look into setFitToScreenOnLayout for GA.